openmc/mesh.py

```diff
@@ -84,6 +84,32 @@ def from_hdf5(cls, group):
         else:
             raise ValueError('Unrecognized mesh type: "' + mesh_type + '"')
 
+    @classmethod
+    def from_xml(cls, elem):
+        """Generates a mesh from an XML element
+
+        Parameters
+        ----------
+        elem : xml.etree.ElementTree.Element
+            XML element
+
+        Returns
+        -------
+        openmc.MeshBase
+            an openmc mesh object
+
+        """
+        mesh_type = get_text(elem, 'type')
+
+        if mesh_type == 'regular' or mesh_type is None:
+            return RegularMesh.from_xml_element(elem)
+        elif mesh_type == 'rectilinear':
+            return RectilinearMesh.from_xml_element(elem)
+        elif mesh_type == 'unstructured':
+            return UnstructuredMesh.from_xml_element(elem)
```

Comment:
The documentation provided here says any mesh type is allowed, but we also have cylindrical and spherical mesh types to handle after the latest PR into OpenMC. Should either handle them here, or update the documentation to warn that spherical and cyl. meshes can't be used with WWs.

Reply:
Fair point, but at the moment the `SphericalMesh` and `CylindricalMesh` objects don't have `from_xml_element` methods (we failed to request them in our review of #1931). If you don't mind, I'll make an issue for this and update this method in a subsequent PR.

Reply:
Issue: #1951